packages/clerk-js/src/ui/common/verifyMagicLink/VerifyMagicLink.tsx

```diff
@@ -18,20 +19,21 @@ export function VerifyMagicLink({
 }: VerifyMagicLinkProps): JSX.Element | null {
   const { handleMagicLinkVerification } = useCoreClerk();
   const { navigate } = useNavigate();
-
+  const signUp = useCoreSignUp();
+  const { displayConfig } = useEnvironment();
   const [verificationStatus, setVerificationStatus] = useState<VerificationStatus>('loading');
 
   useEffect(() => {
     const verify = async () => {
       try {
-        await handleMagicLinkVerification(
-          {
-            redirectUrlComplete,
-            redirectUrl,
-          },
-          navigate,
-        );
+        await handleMagicLinkVerification({ redirectUrlComplete, redirectUrl }, navigate);
         setVerificationStatus('verified');
+
+        return CompleteSignUpFlow({
+          signUp,
+          signUpUrl: displayConfig.signUpUrl,
```

Comment:
⛔ @chanioxaris urls should always be fetched from the useCoreSignUp or useCoreSignIn contexts as that's where we compute their final value based on props and Dashboard configuration. In cases where a component such as VerifyMagicLink is shared across multiple ClerkJS components, make the signUpUrl a component prop and pass its value based on the SignUp Or SignIn context.